### ISSUE TITLE
Consider STB as a replacement for libpng

### ISSUE THREAD

**Issue by santorac** [o3de/o3de#4121]
Consider stb's `stb_image_write.h` as a smaller replacement for libpng.

libpng was introduced as a replacement for OpenImageIO (at least in the runtime). stb was not considered at the time this decision was made, and was suggested during PR. 

_Originally posted by @nemerle in https://github.com/o3de/o3de/issues/4006#issuecomment-915940117_

**Comment by lawsonamzn** [comment]
Yes, assuiming the license for STB is good (looks like its PUBLIC DOMAIN OR MIT) we can grab pieces of it and use it.  

It would NOT make a good tool-side library for image processing, but its intended purpose is a quick dump from runtime with no dependencies, which is great.

**Comment by wintermute-motherbrain** [comment]
sig/graphics-audio approved this RFC on 2021/10/20
